### ISSUE TITLE
By 'docker info' command, 'Dirs' parameter is too big 

### ISSUE THREAD

**Issue by penglongli** [moby/moby#35917]
### Description

**Step 1**

When i execute `docker info`, this command take a long time.
```shell
root@10-10-69-232:/# time docker info &>/dev/null

real	0m4.247s
user	0m0.010s
sys	0m0.009s
```
Contrast with `docker ps -a`, it only uses a quickly time
```shell
root@10-10-69-232:/# time docker ps &>/dev/null

real	0m0.024s
user	0m0.013s
sys	0m0.009s
```

**Step 2**

And then, i executed `docker info` on this machine
```shell
root@10-10-69-232:/# docker info
Containers: 2
 Running: 2
 Paused: 0
 Stopped: 0
Images: 2
Server Version: 17.12.0-ce
Storage Driver: aufs
 Root Dir: /data/docker/aufs
 Backing Filesystem: extfs
 Dirs: 591056
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 89623f28b87a6004d4b785663257362d1658a729
runc version: b2567b37d7b75eb4cf325b77297b140ea686ce8f
init version: 949e6fa
Security Options:
 apparmor
Kernel Version: 3.13.0-137-generic
Operating System: Ubuntu 14.04.5 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.798GiB
Name: 10-10-69-232
ID: OAAK:2NS6:PQMV:7ZSF:LT4F:EU6Y:75XQ:TQWZ:PV5H:BMLS:2RPY:X7ZV
Docker Root Dir: /data/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Registry Mirrors:
 https://registry.docker-cn.com/
Live Restore Enabled: false
```

I found that `Dirs` parameter is too big

**Step 3**

But, i had clean up no-usage `image`、`container`. Below is the output
```shell
root@10-10-69-232:/# docker ps -a
CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS              PORTS               NAMES
d399373f273a        prom/node-exporter:v0.14.0   "/bin/node_exporter"     9 hours ago         Up 44 minutes                           node-exporter-26_1514916118
1c45b7239c80        google/cadvisor:v0.27.2      "/usr/bin/cadvisor -…"   10 hours ago        Up 44 minutes                           cadvisor-49_1514914360
root@10-10-69-232:/# docker images -a
REPOSITORY           TAG                 IMAGE ID            CREATED             SIZE
google/cadvisor      v0.27.2             aee3090515da        2 months ago        59.2MB
prom/node-exporter   v0.14.0             bb4a6b774658        9 months ago        18.9MB
```

### What i expected

I do not know much about the storage structure of `Docker`, and i hadn't find some docs can help me.

What i expected is below:
- I want reduce the size of the `Dirs` para. So `docker info` command will be soon.
- If there has some links about 'Docker storage structure', i would be greatly appreciated

### Environment Info

Machine
```
root@10-10-69-232:/# uname -a
Linux 10-10-69-232 3.13.0-137-generic #186-Ubuntu SMP Mon Dec 4 19:09:19 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
root@10-10-69-232:/#
root@10-10-69-232:/# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty
```

Docker version
```
root@10-10-69-232:/# docker version
Client:
 Version:	17.12.0-ce
 API version:	1.35
 Go version:	go1.9.2
 Git commit:	c97c6d6
 Built:	Wed Dec 27 20:10:36 2017
 OS/Arch:	linux/amd64

Server:
 Engine:
  Version:	17.12.0-ce
  API version:	1.35 (minimum version 1.12)
  Go version:	go1.9.2
  Git commit:	c97c6d6
  Built:	Wed Dec 27 20:09:12 2017
  OS/Arch:	linux/amd64
  Experimental:	false
```

Docker Opts (/etc/default/docker, Ubuntu 14.04 LTS)
```
DOCKER_OPTS="-H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock -g /data/docker
```

Docker Info is at `Step 2` on this ISSUE.

###

**Comment by thaJeztah** [comment]
Were you previously running an older version of Docker? If so; have you used `docker rm --force` to remove containers? If so, it's possible that docker removed containers, but failed to remove their filesystem if the filesystem was in use.

Recent versions of docker in that situation will refuse to remove the container, so that it's possible to re-try removing the container. However, if you've ran an older version of docker on that machine, old directories will not be cleaned up.

If you don't have anything important (i.e., no images, containers or volumes that you have no backup of), you can do a "factory reset", by;

- stopping the docker service
- after it's stopped, remove the `/var/lib/docker` directory (when using the default configuration); it looks like you're using a custom storage-directory, so in that case, you'll have to empty the `/data/docker` directory.
- start the docker service again

**The steps above will remove all images, containers, and volumes from your machine** so only do so if you have nothing important.


I'd also recommend upgrading your kernel to a more recent version; you can install the updated 4.x kernels (from the Xenial release) when running Ubuntu 14.04. Using a 4.x kernel also allows you to use the overlay2 storage driver instead of aufs

**Comment by penglongli** [comment]
@thaJeztah Thanks for your reply!

Yeah, i ran an older version of Docker on this machine previously. I did exactly that used `docker rm --force` to remove containers.

When in the old Docker version, data dir is `/var/lib/docker`. Then i upgrade Docker version, and change data dir to `/data/docker`.  In order to ensure that the old data is not lost, i `mv /var/lib/docker` to `/data/docker`. Now it seems that what i do is stupid.

I have followed your advice, done a "factory reset". It looks normal now. And i'll learn some about Docker Storage.

Thank you again. This ISSUE will be closed.
